slick/src/main/scala/slick/lifted/ExtensionMethods.scala

```diff
@@ -58,6 +58,23 @@ trait ColumnExtensionMethods[B1, P1] extends Any with ExtensionMethods[B1, P1] {
     Library.IfNull.column[P2](n, e.toNode)(tpe(e))
 }
 
+final class Tuple2ColumnExtensionMethods[A: TypedType, B: TypedType](val c: Rep[(A, B)]) extends BaseExtensionMethods[(A, B)] {
+  def inSet[R](seq: Iterable[(A, B)])(implicit om: o#to[Boolean, R]) =
+    if(seq.isEmpty) om(LiteralColumn(false))
+    else {
+      val pairs = seq.map(v => ProductNode(ConstArray(LiteralNode(implicitly[TypedType[A]], v._1), LiteralNode(implicitly[TypedType[B]], v._2))))
+      om.column(Library.In, n, ProductNode(ConstArray.from(pairs)))
+    }
+}
+
+final class Tuple2OfColumnExtensionMethods[A: TypedType, B: TypedType](val t: (Rep[A], Rep[B]))(implicit shapeA: Shape[_ <: FlatShapeLevel, Rep[A], A, Rep[A]], shapeB: Shape[_ <: FlatShapeLevel, Rep[B], B, Rep[B]])
```

Comment:
this is the class that exposes `(Rep[A], Rep[B]) inSet Set((a1, b1), (a2, b2), ...)`

Reply:
could also combine the two classes into a single one;

```scala
final class Tuple2OfColumnExtensionMethods[A: TypedType, B: TypedType](val t: (Rep[A], Rep[B]))(implicit shapeA: Shape[_ <: FlatShapeLevel, Rep[A], A, Rep[A]], shapeB: Shape[_ <: FlatShapeLevel, Rep[B], B, Rep[B]]) {
  def inSet[R](seq: Iterable[(A, B)])(implicit om: OptionMapperDSL.arg[(A, B), (A, B)]#to[Boolean, R]) = {
    if(seq.isEmpty) om(LiteralColumn(false))
    else {
      val n = ProductNode(ConstArray(t._1.toNode, t._2.toNode))
      val pairs = seq.map(v => ProductNode(ConstArray(LiteralNode(implicitly[TypedType[A]], v._1), LiteralNode(implicitly[TypedType[B]], v._2))))
      om.column(Library.In, n, ProductNode(ConstArray.from(pairs)))
    }
  }
}
```